community-connectors/google-fit/GoogleFit.js

```diff
@@ -0,0 +1,255 @@
+
+      // Set the name of the callback function in the script referenced
+      // above that should be invoked to complete the OAuth flow.
+      .setCallbackFunction('authCallback')
+
+      // Set the property store where authorized tokens should be persisted.
+      .setPropertyStore(PropertiesService.getUserProperties())
+
+      // Set the scopes to request (space-separated for Google services).
+      .setScope('https://www.googleapis.com/auth/fitness.activity.read https://www.googleapis.com/auth/fitness.body.read')
+
+      // Below are Google-specific OAuth2 parameters.
+
+      // Sets the login hint, which will prevent the account chooser screen
+      // from being shown to users logged in with multiple accounts.
+      .setParam('login_hint', Session.getActiveUser().getEmail())
+
+      // Requests offline access.
+      .setParam('access_type', 'offline')
+
+      // Forces the approval prompt every time. This is useful for testing,
+      // but not desirable in a production application.
+      .setParam('approval_prompt', 'force');
+
+  return this._service;
+};
+
+
+
+/**
+ * Gets activity for the signed in user during the given time period.
+ *
+ * @param{Date} startTime the start time for the period to get activity
+ * @param{Date} endTime the end time for the period to get activity
+ * @return {Object} a Google Fit activity API JSON response.
+*/
+GoogleFit.prototype.getActivity = function(startTime, endTime) {
+  return this._getDatasets('derived:com.google.activity.segment:com.google.android.gms:merge_activity_segments', startTime, endTime);
+};
+
+/**
+ * Gets steps for the signed in user during the given time period.
+ *
+ * @param{Date} startTime the start time for the period to get steps
+ * @param{Date} endTime the end time for the period to get steps
+ * @return {Object} a Google Fit steps API JSON response.
+*/
+GoogleFit.prototype.getSteps = function(startTime, endTime) {
+  return this._getDatasets('derived:com.google.step_count.delta:com.google.android.gms:merge_step_deltas', startTime, endTime);
+};
+
+/**
+ * Gets weight for the signed in user during the given time period.
+ *
+ * @param{Date} startTime the start time for the period to get weight 
+ * @param{Date} endTime the end time for the period to get weight 
+ * @return {Object} a Google Fit weight API JSON response.
+*/
+GoogleFit.prototype.getWeight = function(startTime, endTime) {
+  return this._getDatasets('derived:com.google.weight:com.google.android.gms:merge_weight', startTime, endTime);
+};
+
+GoogleFit.prototype._getDatasets = function(dataSource, startTime, endTime) {
+  // TODO: Implement paging using pageToken
+  //       See: https://developers.google.com/fit/rest/v1/reference/users/dataSources/datasets/get
+  var service = this.getService();
+  var uri = 'https://www.googleapis.com/fitness/v1/users/me/dataSources/' + dataSource + '/datasets/' + (startTime.getTime() * 1000000) + '-' + (endTime.getTime() * 1000000);
```

Comment:
It'd be good to pull out the magic numbers such as `1000000` into named variables.

Reply:
Done